l10n_ch_payment_slip/tests/test_payment_slip.py

```diff
@@ -136,6 +136,62 @@ def test_slip_validity(self):
                 )
                 self.assertIn(line_ident, slip.reference.replace(' ', ''))
 
+    def test_isr_reference(self):
+        # no partner ref
+        self.env.ref('base.res_partner_12').ref = ''
+        invoice = self.make_invoice()
+        self.assertTrue(invoice.isr_reference)
+        reference = invoice.isr_reference.replace(' ', '')
+        self.assertEqual(len(reference), 27)
+        self.assertEqual(reference[:6], '123456')
+        self.assertEqual(reference[6:13], '0' * 7)
+
+        # standard
+        self.env.ref('base.res_partner_12').ref = '12345'
+        invoice = self.make_invoice()
+        self.assertTrue(invoice.isr_reference)
+        reference = invoice.isr_reference.replace(' ', '')
+        self.assertEqual(len(reference), 27)
+        self.assertEqual(reference[:6], '123456')
+        self.assertEqual(reference[6:13], '0012345')
+
+        # partner ref without num
+        self.env.ref('base.res_partner_12').ref = 'alpha'
+        invoice = self.make_invoice()
+        self.assertTrue(invoice.isr_reference)
+        reference = invoice.isr_reference.replace(' ', '')
+        self.assertEqual(len(reference), 27)
+        self.assertEqual(reference[:6], '123456')
+        self.assertEqual(reference[6:13], '0' * 7)
+
+        # contains numerics
```

Comment:
```suggestion
        # contains alphabetic
```

Reply:
Here for me the important part is 123 which will become 0000123.
Thus I meant it contains this but not only.

Will change to alphanumeric then.

Reply:
```suggestion
        # contains alphanumerics
```